src/Medidata.CrossApplicationTracer/ZipkinSampler.cs

```diff
@@ -57,9 +69,9 @@ public virtual bool ShouldBeSampled(System.Web.HttpContextBase httpContext, stri
                 return result;
             }
 
-            if ( ! IsInDontSampleList(httpContext.Request.Path))
+            if (!IsInDontSampleList(httpContext.Request.Path))
             {
-                if ( random.NextDouble() <= sampleRate )
+                if (random.NextDouble() <= sampleRate)
```

Comment:
Do you know by any chance why we need to randomize the ShouldBeSampled result here? I understand that if this is not defined by the caller it will be randomly sampled or not, but I don't understand why we need to put an indeterministic factor into the implementation.